CONTRIBUTING.md

```diff
@@ -99,3 +102,141 @@ Although we greatly appreciate any and all contributions to the project, there a
+    - Open Settings by pressing ***Ctrl+Alt+S***
+      ![](images/open-settings-4.1.png)
+
+    - Follow the path ***IDE settings>Language & Frameworks > Lombok*** and then click on ***Lombok.***
+      ![](images/lombok-setup-4.2.png)
+
+    - Make sure that the ***Lombok processing*** is enabled.
+      ![](images/lombok-processing-4.3.png)
+
+- ### Step 5: Reload the project
+    - Open the ***Maven*** Tab
+    
+      ![](images/open-maven-5.1.png)
+
+    - Press the ***Reload*** button as shown below and allow the project to Reload.
+    
+      ![](images/reload-maven-5.2.png)
+
+- ### Step 6: Running the Project.
+    - Open the ***WrongSecretsApplication*** by following the path ***main>java>org.owasp.wrongsecrets>WrongSecretApplication***.
+      ![](images/open-application-6.1.png)
+    - Press ***Shift+F10*** to run the application, this will open up the ***Run/Debug Configurations Menu.***
+      ![](images/run-application-6.2.png)
+
+- ### Step 7: Setting up Configurations.
+    -  Select ***Edit configuration templates*** then select ***Application*** section.
+    
+       ![](images/edit-config-7.1.png)
+    - There under the ***Application*** section click on the button shown below.
+    
+      ![](images/modify-options-7.2.png)
+    - ***Select*** all the fields that are Selected in the below picture.
+    
+      ![](images/select-options-7.3.png)
+    - ***Fill*** all the fields as shown below.
+    
+      ![](images/fill-fields-7.4.png)
+    - Again press ***Shift+F10*** which runs the Application.
+    
+      ![](images/run-application-6.2.png)
+
+- ### There you have it, ***WrongSecrets*** running successfully.
+    - Here is a *preview* on how does it look after successfully running the Application.
+      **Note:** Running the Application doesn't open any kind of ***GUI***, it only initializes the ***local webserver*** that you can open via a ***browser.***
+      ![](images/final-output-8.png)
+
+    - Here is the preview of the **web server**,  you can try to find the secrets by means of solving the challenge offered at:
+      [**Challenges**](https://github.com/OWASP/wrongsecrets#basic-docker-exercises)
+        ![](images/screenshot.png)
+
+---
+## How to add challenge
+- ### Step 1: Creating a new issue.
+  First make sure that you have an [Issue](https://github.com/OWASP/wrongsecrets/issues/new) reported for which a challenge is really wanted.
+
+- ### Step 2: Adding the challenge.
+  Add the **new challenge** in this folder `wrongsecrets/src/main/java/org/owasp/wrongsecrets/challenges/`.
+  These are the things that you have to keep in mind.
+    - First and foremost make sure your challenge is coded in **Java**.
+    - Don't forget to add your challenge number in `@Order(28)` annotation, ***28*** in my case.
+    - For further reference please refer [*Challenge28.java*](https://github.com/puneeth072003/wrongsecrets/blob/Readme-updation/src/main/java/org/owasp/wrongsecrets/challenges/docker/Challlenge28.java)
```

Comment:
```suggestion
Here is an example of a possible Challenge 28:
```java
package org.owasp.wrongsecrets.challenges.docker;


import lombok.extern.slf4j.Slf4j;
import org.owasp.wrongsecrets.RuntimeEnvironment;
import org.owasp.wrongsecrets.ScoreCard;
import org.owasp.wrongsecrets.challenges.Challenge;
import org.owasp.wrongsecrets.challenges.ChallengeTechnology;
import org.owasp.wrongsecrets.challenges.Spoiler;
import org.springframework.core.annotation.Order;
import org.springframework.stereotype.Component;

import java.util.List;

@Slf4j
@Component
@Order(28) //make sure this number is the same as your challenge
public class Challenge28 extends Challenge {
    private final String secret;

    public Challenge28(ScoreCard scoreCard) {
        super(scoreCard);
        secret = "hello world";
    }

    //is this challenge usable in CTF mode?
    @Override
    public boolean canRunInCTFMode() {
        return true;
    }

    //return the plain text secret here
    @Override
    public Spoiler spoiler() {
        return new Spoiler(secret);
    }


    //here you validate if your answer matches the secret
    @Override
    public boolean answerCorrect(String answer) {
        return secret.equals(answer);
    }

    //which runtime can you use to run the challenge on ? (You can just use Docker here)
    @Override
    public List<RuntimeEnvironment.Environment> supportedRuntimeEnvironments() {
        return List.of(RuntimeEnvironment.Environment.DOCKER);
    }

    //set the difficulty: 1=low, 5=very hard
    @Override
    public int difficulty() {
        return 1;
    }

    //on which tech is this challenge? See ChallengeTechnology.Tech for categories
    @Override
    public String getTech() {
        return ChallengeTechnology.Tech.SECRETS.id;
    }

    //if you use this in a shared environment and need to adapt it, then return true here.
    @Override
    public boolean isLimittedWhenOnlineHosted() {
        return false;
    }

}
```
```

Reply:
@commjoen Should i remove challenge 28?

Reply:
Yes, just apply the above please